src/coreclr/gc/gc.cpp

```diff
@@ -37780,25 +38261,43 @@ void gc_heap::trim_youngest_desired_low_memory()
     }
 }
 
+ptrdiff_t gc_heap::estimate_gen_growth (int gen)
+{
+    dynamic_data* dd_gen = dynamic_data_of(gen);
+
+    ptrdiff_t new_allocation_gen = dd_new_allocation (dd_gen);
+    ptrdiff_t free_list_space_gen = generation_free_list_space (generation_of (gen));
+
+    // estimate how we are going to need in this generation - estimate half the free list space gets used
+    ptrdiff_t budget_gen = new_allocation_gen - (free_list_space_gen / 2);
+    dprintf (REGIONS_LOG, ("budget for gen %d on heap %d is %Id (new %Id, free %Id)", gen, heap_number, budget_gen, new_allocation_gen, free_list_space_gen));
+    if (budget_gen > 0)
+    {
+        return budget_gen;
+    }
+    else
+    {
+        return 0;
+    }
+}
```

Comment:
could just be ` return max (0, budget_gen)`?

Reply:
Ok, done.